source/slang/slang-lower-to-ir.cpp

```diff
@@ -1653,6 +1656,86 @@ struct ValLoweringVisitor : ValVisitor<ValLoweringVisitor, LoweredValInfo, Lower
         return LoweredValInfo::simple(resultVal);
     }
 
+    LoweredValInfo visitConcreteTypePack(ConcreteTypePack* typePack)
+    {
+        ShortList<IRType*> types;
```

Comment:
I am not sure what the max value can be from getTypeCount.
Maybe we should reserve space fore `types` before start adding elements to it.

Reply:
We can, but this kind of things doesn't really matter that much. Shortlist is already by default reserved to be 16 in length and very few tuples in practice would exceed that anyways. If they do the next allocation would be 32.